src/hooks/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Dropdown, DropdownItem, DropdownToggle, ToolbarExpandIconWrapper, TreeView, TreeViewDataItem } from "@patternfly/react-core";
+import { AngleDownIcon } from "@patternfly/react-icons";
+import { ColumnsMetada, usePrimaryToolbarFilterConfig } from "@redhat-cloud-services/insights-common-typescript";
+import React from "react";
+import { useEffect, useMemo } from "react"
+import { ClearEventLogFilters, EventLogFilterColumn, EventLogFilters, SetEventLogFilters } from "../components/Notifications/EventLog/EventLogFilter";
+import { Schemas } from "../generated/OpenapiNotifications";
+
+interface TreeFilterProps {
+    groups: readonly Schemas.Facet[],
+    items: readonly Schemas.Facet[],
+    placeholder: string
+}
+
+export const TreeDropdownFilter: React.FunctionComponent<TreeFilterProps> = (props) => {
+    const [isFocused, setIsFocused] = React.useState(true)
+    const data: TreeViewDataItem[] = props.groups.map(group => ({
+        id: group.name,
+        name: group.displayName,
+        checkProps: { checked: false },
+        children: props.items.map(item => ({
+            id: item.name,
+            name: item.displayName,
+            checkProps: { checked: false }
+        }))
+
+    }))
+
+    console.log("render")
+    return (
+        <Dropdown
+            onSelect={() => setIsFocused(!isFocused)}
```

Comment:
I don't think you want this here. 
The `onSelect` is used to know when you clicked something from the content.
Since you are going to use a tree-view; that component should handle their own selection/clicks.

```suggestion
```